server/helpers.go

```diff
@@ -0,0 +1,201 @@
+	m.Meta = description
+	var RedirectUrl *url.URL
+	RedirectUrl, _ = url.Parse(callbackURL)
+	RedirectUrl.Path += "/plugins/bigbluebutton/redirect"
+	StringRedirectUrl := RedirectUrl.String()
+	m.LogoutURL = StringRedirectUrl
+	m.LoopCount = 0
+	m.ValidToken = GenerateRandomID()
+
+	var recordingcallbackurl string
+	var Url *url.URL
+	Url, _ = url.Parse(callbackURL)
+	Url.Path += "/plugins/bigbluebutton/recordingready"
+	recordingcallbackurl = Url.String()
+	m.Meta_bn_recording_ready_url = recordingcallbackurl
+
+	var UrlEnd *url.URL
+	UrlEnd, _ = url.Parse(callbackURL)
+	UrlEnd.Path += "/plugins/bigbluebutton/meetingendedcallback?" + m.MeetingID_ + "&" + m.ValidToken
+	Endmeetingcallback := UrlEnd.String()
+	m.Meta_endcallbackurl = Endmeetingcallback
+	return nil
+}
+
+func (p *Plugin) LoadMeetingsFromStore() {
+	byted, _ := p.API.KVGet("all_meetings")
+	json.Unmarshal(byted, &p.Meetings)
+
+	recordings_byted, _ := p.API.KVGet("recording_queue")
+	json.Unmarshal(recordings_byted, &p.MeetingsWaitingforRecordings)
+
+}
+func (p *Plugin) SaveMeetingToStore() {
+	byted, _ := json.Marshal(p.Meetings)
+	p.API.KVSet("all_meetings", byted)
+
+	recordings_byted, _ := json.Marshal(p.MeetingsWaitingforRecordings)
+	p.API.KVSet("recording_queue", recordings_byted)
+
+}
+
+// Returns a meeting pointer so we'll be able to manipulate its content from outside the array.
+func (p *Plugin) FindMeeting(meeting_id string) *dataStructs.MeetingRoom {
+	for i := range p.Meetings {
+		if p.Meetings[i].MeetingID_ == meeting_id {
+			return &(p.Meetings[i])
+		}
+	}
+	return nil
+}
+func (p *Plugin) FindMeetingfromInternal(meeting_id string) *dataStructs.MeetingRoom {
+	for i := range p.Meetings {
+		if p.Meetings[i].InternalMeetingId == meeting_id {
+			return &(p.Meetings[i])
+		}
+	}
+	return nil
+}
+
+
+func (p *Plugin) createStartMeetingPost(user_id string, channel_id string, m *dataStructs.MeetingRoom) {
+
+	config := p.config()
+	// If config page is not set uh oh.
+	if err := config.IsValid(); err != nil {
+		return
+	}
+
+	textPost := &model.Post{UserId: user_id, ChannelId: channel_id,
+		Message: "#BigBlueButton #" + m.Name_ + " #ID" + m.MeetingID_, Type: "custom_bbb"}
+
+	textPost.Props = model.StringInterface{
+		"from_webhook":      "true",
+		"override_username": "BigBlueButton",
+		"override_icon_url": "https://pbs.twimg.com/profile_images/467451035837923328/JxPpOTL6_400x400.jpeg",
+		"meeting_id":        m.MeetingID_,
+		"meeting_status":    "STARTED",
+		"meeting_personal":  false,
+		"meeting_topic":     m.Name_, // Fill in this meeting topic.
+		"meeting_desc":      m.Meta,
+		"user_count":        0,
+	}
+
+	postpointer, _ := p.API.CreatePost(textPost)
+	m.PostId = postpointer.Id
+}
+
+func (p *Plugin) DeleteMeeting(meeting_id string) {
+	var index int
+	for i := range p.Meetings {
+		if p.Meetings[i].MeetingID_ == meeting_id {
+			index = i
+			break
+		}
+	}
+	p.Meetings = append(p.Meetings[:index], p.Meetings[index+1:]...)
```

Comment:
You shouldn't be keeping data in-memory. This can be easily lost on plugin crash. Moreover, the plugin may not work correctly when used in high availability mode. https://developers.mattermost.com/extend/plugins/server/ha/ .

Reply:
Filed #61 as this is a longer work.